tests/template_tests/test_extends.py

```diff
@@ -181,3 +182,27 @@ def test_block_override_in_extended_included_template(self):
         )
         template = engine.get_template("base.html")
         self.assertEqual(template.render(Context({})), "12AB")
+
+    def test_extends_not_first_tag_in_extended_template(self):
+        engine = Engine(
+            loaders=[
+                [
+                    "django.template.loaders.locmem.Loader",
+                    {
+                        "base.html": (
+                            '{% extends "base.html" %}',
+                            "{% block content %}{% endblock content %}",
+                        ),
+                        "index.html": (
+                            "{% block content %}B{% endblock %}"
+                            "{% extends 'base.html' %}"
+                        ),
+                    },
+                ],
+            ],
+        )
+        msg = "{% extends 'base.html' %} must be the first tag in \"index.html\"."
+
+        with self.assertRaisesMessage(TemplateSyntaxError, msg):
+            template = engine.get_template("index.html")
+            template.render(Context({}))
```

Comment:
Updating the test to add the render_from_string case
```suggestion
        bad_template = "{% block content %}B{% endblock %}{% extends 'base.html' %}"
        engine = Engine(
            loaders=[
                [
                    "django.template.loaders.locmem.Loader",
                    {
                        "base.html": (
                            '{% extends "base.html" %}',
                            "{% block content %}{% endblock content %}",
                        ),
                        "index.html": bad_template,
                    },
                ],
            ],
        )
        msg = "{% extends 'base.html' %} must be the first tag in \"index.html\"."
        with self.subTest(msg=msg):
            with self.assertRaisesMessage(TemplateSyntaxError, msg):
                engine.get_template("index.html")

        msg = "{% extends 'base.html' %} must be the first tag in the template."
        with self.subTest(msg=msg):
            with self.assertRaisesMessage(TemplateSyntaxError, msg):
                engine.from_string(bad_template)
```

Reply:
not sure about giving the `index.html` content from a string, because engine still think it's coming from `index.html`. Maybe we can add another test like below, also it's passing with your suggestion.

```python
    def test_extends_not_first_tag_in_extended_template_from_string(self):
        engine = Engine()
        template_string = "{% block content %}B{% endblock %}" "{% extends 'base.html' %}"
        msg = "{% extends 'base.html' %} must be the first tag in the template."

        with self.assertRaisesMessage(TemplateSyntaxError, msg):
            template = engine.from_string(template_string)
            template.render(Context({}))
```

how is that looks ?

Reply:
Looks great thank you